src/power_grid_model/core/buffer_handling.py

```diff
@@ -64,7 +91,23 @@ def _get_raw_data_view(data: np.ndarray, schema: ComponentMetaData) -> VoidPtr:
     Returns:
         a raw view on the data set.
     """
-    return np.ascontiguousarray(data, dtype=schema.dtype).ctypes.data_as(VoidPtr)
+    if isinstance(data, np.ndarray):
+        return _get_raw_data_view(data, dtype=schema.dtype)
+    return None
+
+
+def _get_raw_attribute_data_view(data: np.ndarray, schema: ComponentMetaData, attribute: AttributeType) -> VoidPtr:
+    """
+    Get a raw view on the data.
+
+    Args:
+        data: the data.
+        schema: the schema the raw buffer should obey.
+
+    Returns:
+        a raw view on the data set.
+    """
+    return _get_raw_data_view(data, dtype=schema.dtype[attribute])
```

Comment:
```suggestion
    return _get_raw_data_view(data, dtype=schema.dtype[attribute].base)
```
This caused asym tests to fail.